x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/EsqlParser.java

```diff
@@ -102,27 +121,31 @@ public void syntaxError(
             String message,
             RecognitionException e
         ) {
+            if (recognizer instanceof EsqlBaseParser parser && parser.isDevVersion() == false) {
+                Matcher m = REPLACE_DEV.matcher(message);
+                message = m.replaceAll(StringUtils.EMPTY);
+            }
+
```

Comment:
Hacky way to remove the DEV_ tokens - if anybody else has better ideas, let me know.
FTR, I've tried removing the token from the stream, moving them to a hidden channel however it doesn't work since the error is generated by the recognizer who's inspecting the ATN (the transition between states) and the DEV_ token show up. 
The code above is much easier than manipulating the ATN, which gets generated by ANTLR - though it's probably quite fragile.